File: migrate/test/fixtures/bot.input.py
import pmxt

venue = pmxt.Hyperliquid(
    pmxt_api_key="pmxt_live_xxx",
    wallet_address="0xYourWalletAddress",
    private_key="0xYourPrivateKey",
)

kalshi = pmxt.Kalshi()


def main():
    markets = venue.fetch_markets(query="BTC", limit=10)
    outcome_id = markets[0].outcomes[0].outcome_id

    book = venue.fetch_order_book(outcome_id)
    print("best bid", book.bids[0].price, "best ask", book.asks[0].price)

    candles = venue.fetch_ohlcv(outcome_id, resolution="1h", limit=100)
    print("last close", candles[-1].close)

    balances = venue.fetch_balance()
    print(balances)

    try:
        order = venue.create_order(
            market_id=markets[0].market_id,
            outcome_id=outcome_id,
            side="buy",
            order_type="limit",
            amount=5,
            price=0.42,
        )
        venue.cancel_order(order.id)
    except pmxt.MarketNotFound:
        print("unknown market")
    except pmxt.InsufficientFunds:
        print("top up")

    orders = venue.fetch_all_orders(market_id=markets[0].market_id, limit=50)
    print(orders)

    venue.close()


main()

File: migrate/test/fixtures/bot.input.ts
import { Hyperliquid, Polymarket, MarketNotFound, InsufficientFunds } from "pmxtjs";

const venue = new Hyperliquid({
    pmxtApiKey: process.env.PMXT_API_KEY,
    walletAddress: process.env.WALLET_ADDRESS,
    privateKey: process.env.PRIVATE_KEY,
});

const poly = new Polymarket({ pmxtApiKey: process.env.PMXT_API_KEY });

async function main () {
    const markets = await venue.fetchMarkets({ limit: 10, sort: "volume" });
    const outcomeId = markets[0].outcomes[0].outcomeId;

    const book = await venue.fetchOrderBook(outcomeId);
    console.log("best bid", book.bids[0].price, "best ask", book.asks[0].price);

    const candles = await venue.fetchOHLCV(outcomeId, "1h", 100);
    console.log("last close", candles[candles.length - 1].close);

    const balances = await venue.fetchBalance();
    console.log(balances);

    try {
        const order = await venue.createOrder({
            marketId: markets[0].marketId,
            outcomeId: outcomeId,
            side: "buy",
            type: "limit",
            amount: 5,
            price: 0.42,
        });
        await venue.cancelOrder(order.id);
    } catch (e) {
        if (e instanceof MarketNotFound) {
            console.error("unknown market");
        } else if (e instanceof InsufficientFunds) {
            console.error("top up");
        }
    }

    const edge = await venue.fetchArbitrage({ minSpread: 0.02 });
    console.log(edge);

    await venue.close();
}

main();
